File: Stage9/C/Problem1978.c
#include <stdio.h>

int main(void) {
    int N;
    scanf("%d", &N);

    int count = 0;

    for (int i = 0; i < N; i++) {
        int num;
        scanf("%d", &num);

        if (num <= 1) continue;

        int isPrime = 1;
        for (int j = 2; j * j <= num; j++) {
            if (num % j == 0) {
                isPrime = 0;
                break;
            }
        }
        if (isPrime) count++;
    }
    
    printf("%d\n", count);
    return 0;
}

File: Stage9/Java/Problem1978.java
package Stage9.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Problem1978 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        StringTokenizer st = new StringTokenizer(br.readLine());
        int count = 0;

        for (int i = 0; i < N; i++) {
            int num = Integer.parseInt(st.nextToken());
            if (num > 1) {
                boolean isPrime = true;
                for (int j = 2; j * j <= num; j++) {
                    if (num % j == 0) {
                        isPrime = false;
                        break;
                    }
                }
                if (isPrime) count++;
            }
        }

        System.out.println(count);
        br.close();
    }
}
